includes/class-wc-session-handler.php

```diff
@@ -253,6 +256,9 @@ public function save_data() {
 
 			wp_cache_set( $this->get_cache_prefix() . $this->_customer_id, $this->_data, WC_SESSION_CACHE_GROUP, $this->_session_expiration - time() );
 			$this->_dirty = false;
+			if ( get_current_user_id() != $old_session_key ) {
```

Comment:
There is a slight problem I see with this approach. I think the `$guest_session_id` from above (`$old_session_key` in this context) is not always coming from the guest. If I'm logged in as User_1 and go to login screen and then log in as User_2 (without the logout step), session of User_1 gets deleted. 

Perhaps that's not a very common thing to do for users on an eshop, users could e.g. have a business account separate from their personal account and switch between those? 

I'm not sure about the best way to check this, though. Seems like the guest user's id is an MD5 hash, so maybe check if it has 32 characters and hope someone will fix it before an eshop can have 1x10^32 customers? :)

Reply:
Oh, I didn't know it was possible to login as another user without logging out first. So I'll have to make the parameter optional and perhaps add check for the MD5 hash

```
function isValidMd5($md5 ='') {
  return strlen($md5) == 32 && ctype_xdigit($md5);
}
```

 or use WP function `get_user_by( 'id', $old_session_key )` to check if there is actual user related to the previous session ID as this should return false if the id passed is the guest user MD5 hash https://developer.wordpress.org/reference/functions/get_user_by/.

Do you @rodrigoprimo and @peterfabian think I should prefer one or the other way to check for case of changing from one logged in user to another one? MD5 check or the user object check?

Reply:
Ended up checking for user object with the user ID / guest user hash.